### PR TITLE
Feature/10 add jwt support

### PR DESCRIPTION
**Pull request by OlegMikus**

### CODE REVIEW

**Review by ishaurko — CHANGES_REQUESTED** [review]
1. It is good for readability and maintainability to keep separate class/util in separate .py file
2. It is a good practice to have common structure for response with all fields present all the time i. e. {'data': {}, 'errors': {}, 'items_count': int}

File: src/api/auth.py [hunk 1]
```diff
@@ -1,20 +1,47 @@
...
-    password = str(request_data['password'])
-    user = User.query.filter_by(email=email).first()
-    if not user or not check_password_hash(user.password, password):
-        return jsonify({'info': 'smth wrong'})
+    user = User.query.filter_by(email=request_data['email']).first()
+    if request_data['email'] and check_password_hash(user.password, str(request_data['password'])):
+        session['logged_in'] = True
+
+        access_token = create_access_token(request_data)
+        ref_token = create_refresh_token(request_data)
```
[the remark this answers is not in the document]
Comment (OlegMikus) [reply src/api/auth.py:17]: ok, i'll fix this

File: src/api/auth.py [hunk 2]
```diff
@@ -1,20 +1,47 @@
...
 def login() -> Response:
     request_data = request.get_json()
-    email = request_data['email']
-    password = str(request_data['password'])
-    user = User.query.filter_by(email=email).first()
-    if not user or not check_password_hash(user.password, password):
-        return jsonify({'info': 'smth wrong'})
+    user = User.query.filter_by(email=request_data['email']).first()
+    if request_data['email'] and check_password_hash(user.password, str(request_data['password'])):
+        session['logged_in'] = True
```
[the remark this answers is not in the document]
Comment (OlegMikus) [reply src/api/auth.py:14]: forget to remove this

File: src/api/auth.py [hunk 3]
```diff
@@ -1,20 +1,47 @@
...
+        ref_token = create_refresh_token(request_data)
+
+        return jsonify({'access_token': access_token.decode('UTF-8'),
+                        'refresh_token': ref_token.decode('UTF-8')})
+
+    return jsonify({'info': 'something wrong'})
+
+
+@app.route('/refresh')
+@refresh_required
```
[the remark this answers is not in the document]
Comment (OlegMikus) [reply src/api/auth.py:26]: so, i should change my refresh function to do, what refresh required decorator did, rigth?

**Pull request merged**